### ISSUE TITLE
exitAfterDefer: allow to use "defer stop()" where stop is signal.NotifyContext's stop func

### ISSUE THREAD

**Issue by leonklingele** [go-critic/go-critic#1509]
The `exitAfterDefer` linter reports `exitAfterDefer: os.Exit will exit, and defer sstop() will not run (gocritic)` for the following code:

```go
func main() {
	ctx, sstop := signal.NotifyContext(context.Background(), syscall.SIGINT, syscall.SIGTERM)
	defer sstop()

	if err := run(ctx); err != nil {
		slog.ErrorContext(ctx, "run error", slog.Any("err", err))
		os.Exit(1)
	}
}
```

As we do not need to stop listening for signals anymore when calling `os.Exit`, this construction should not result in a linting error.

**Comment by quasilyte** [comment]
It's not reporting that you need to do something with defer, it merely reports that it will never run if exit is executed (and it's not wrong - it will not be executed). The semantics of a specific deferred call can be tricky to hardcode.

Some main functions are dirty like that - mixing exit and defers and then you can't ever be sure what exactly went wrong with the program shutdown. This checker was made to call out these kinds of functions, as it's usually a code smell.

If you know that this exit/defer pair is safe, a nolint comment with a justification is better in my opinion.

And if you're not feeling on a timer, I usually prefer separating exit-able areas and defer-safe(ish) zones separated by functions.
So only main does exit, but never defers, and any functions called from it can defer, but should usually never do an exit.
```
// main can exit
// runApp can defer
func main() {
  if err := runApp(); err != nil {
    fmt.Println(err)
    os.Exit(1)
  }
}
```

**Comment by leonklingele** [comment]
Thanks, that absolutely makes sense! Better have it very strict and not ignore any such constructions.
